Adapt CI for opensearch 3

Earlopain opened the pull request:
### Description
Add OpenSearch 3 to CI since it is now GA. I made a few commits, see their message for rationale for each.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-ruby/issues/294

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

nhtruong reviewed (dismissed):
This makes sense. Thank you!

dblock commented:
I think we shouldn't lose tests for 1.x, for as long as the client is compatible with that.

Earlopain commented:
> I think we shouldn't lose tests for 1.x, for as long as the client is compatible with that.

Eh, I wouldn't bother personally. I doubt it would actually break in any meaningful way and compatibility.md tells users to use the 1.X opensearch-ruby client version.

For now, I just removed the test against unreleased 1.x versions since that will never happen again and added the other version back to the matrix. It's no big bother.

dblock commented on .github/workflows/test-unreleased.yml:
```diff
@@ -26,7 +26,6 @@ jobs:
       matrix:
         entry:
-          - { ruby_version: '3.3', opensearch_ref: '1.x', jdk_version: '11' }
           - { ruby_version: '3.3', opensearch_ref: '2.x', jdk_version: '17' }
```
No, the 3.0 line collapsed the .x versioning into just having 3.0 = main.
